framework/src/auxkernels/AuxKernel.C

```diff
@@ -108,6 +111,12 @@ AuxKernel::AuxKernel(const InputParameters & parameters) :
   for (const auto & it : coupled_vars)
     for (const auto & var : it.second)
       _depend_vars.insert(var->name());
+
+  if (_displaced_var)
+  {
+    addMooseVariableDependency(_displaced_var);
```

Comment:
I actually mean that if we can resize necessary arrays somewhere, we can get rid of most of extra codes because we are essentially just writing in `AuxKernel`, there is indeed no dependency on the moose variable on the displaced mesh.